### ISSUE TITLE
Inject HELM_RELEASE_NAME and POD_NAMESPACE into the user pods

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/zero-to-jupyterhub-k8s#1747]
### Proposed change
To inject one or two environment variables by default in the user pod, namely `HELM_RELEASE_NAME` and `POD_NAMESPACE` like we do in the JupyterHub pod. This could help any user of the Helm chart configure software on the user pod by referencing the variables.

### Alternative options
To make the helm chart user inject them manually as a separate step.

### Who would use this feature?
A Z2JH helm chart user that wants to make Dask-Gateway easy to use, which requires the user pod to know addresses on where it will be made accessible, but that may require knowledge about its namespace. Hmm... The user pod won't need to know the namespace in this situation though I realize, only the Helm release name (A Helm release name can be thought of a name to the specific installation of the Helm chart into the k8s cluster.)

### (Optional): Suggest a solution
We let the z2jh Helm chart update c.KubeSpawner.environment by merging these environment variables into that configuration after the hub.extraConfig's has been passed and loaded.

**Comment by minrk** [comment]
It seems sensible to me to do this by default, but it's also generally not necessary info for *users* to know what their namespace and helm release name are, since they shouldn't have permission to do anything with that information.

My first ideas:

1. add these by default (maybe some deployments would prefer to hide this info from users?),
2. document how to add these easily (in part to verify if that's even true and try to improve the situation, if not),
3. have an opt-in config option to pass kubernetes/helm config to the user pod, e.g. `singleuser.includeKubernetesEnv = true`, mainly in the event of 2. not being able to be simple enough due to helm template limitations

**Comment by consideRatio** [comment]
Since https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/1757 we have support for using k8s native configuration of environment variables through `singleuser.extraEnv`, so injecting these are now a matter of convenience for administrators I think.

The Helm release name environment variable would only make sense to provide by Z2JH, while the namespace environment variable could be included as part of KubeSpawner's logic if we want. JupyterHub's base spawner class currently provide a lot of JupyterHub environment variables in the [`get_env` function](https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/spawner.py#L727).

I think this kind of convenience could be nice to have standardized for the Helm chart users, but perhaps not so essential. It would also be to expose a bit more information that could be used against the system by a malicious user, so I think it should be opt-in to get it.

In other words, I think I favor option 3 described by @minrk above. I think option 2 was quite nice, but I dislike having to describe how to get the Helm chart release name from a environment variable in the hub pod as it is specific to the Helm chart implementation. Option 2 is fully possible though and I would support that as well.

**Comment by consideRatio** [comment]
With singleuser.extraEnv, I think we should opt for letting users create an env themselves.
